pkgs/development/libraries/arcan/default.nix

```diff
@@ -0,0 +1,185 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake
+, pkg-config
+, xz
+, libuvc
+, libffi
+, libX11
+, libxcb
+, libXcomposite
+, libXfixes
+, xcbutil
+, xcbutilwm
+, libXau
+, libXdmcp
+, openal
+, freetype
+, sqlite
+, libGL
+, libdrm
+, libxkbcommon
+, mesa
+, valgrind
+, luajit
+, libusb1
+, harfbuzzFull
+, SDL2
+
+# For generating API documentation
+, ruby
+
+#
+, withVLC ? true
+, libvlc
+
+#
+, withFFmpeg ? true
+, ffmpeg-full
+
+#
+, withWayland ? true
+, wayland
+, wayland-protocols
+
+#
+, withEspeak ? true
+, espeak-classic
+
+#
+, withLibmagic ? true
+, file
+
+#
+, withTesseractAndLeptonica ? true
+, tesseract
+, leptonica
+
+#
+, withVNC ? true
+, libvncserver
+
+# Output debug information
+, debugging ? false
+}:
+
+let
+  name = "arcan";
+
+  srcs = {
+    arcan = fetchFromGitHub {
+      owner = "letoram";
+      repo = name;
+      rev = "ffc266523bc89def40666cceb17be5707994d1c0";
+      sha256 = "sha256-bfLGjnnxX+sHnDn5eGYKqRR51Gn9al9GEqw9qF63j38=";
+    };
+
+    openal = fetchFromGitHub {
+      owner = "letoram";
+      repo = "openal";
+      rev = "1c7302c580964fee9ee9e1d89ff56d24f934bdef"; # master
+      sha256 = "0dcxcnqjkyyqdr2yk84mprvkncy5g172kfs6vc4zrkklsbkr8yi2";
+    };
+  };
+in stdenv.mkDerivation rec {
+  pname = name;
+
+  version = "0.6.1pre1+unstable=2021-05-31";
+
+  src = srcs.arcan;
+
+  postUnpack = ''
+    mkdir -p $sourceRoot/external/git/openal
+    pushd $sourceRoot/external/git/
+    cp --recursive --verbose ${srcs.openal}/* openal/
+    chmod --recursive 777 openal/
+    popd
+
+    # Generate manpages
+    pushd $sourceRoot/doc/
+    mkdir mantmp
+    ruby docgen.rb mangen
+    popd
```

Comment:
This shouldn't be done in postUnpack. postBuild would be a better fitting phase.

Reply:
Then the build process doesn't have the files, which it want.

Reply:
Then it belongs into preBuild. unpack should not compile anything.